### ISSUE TITLE
ChromeDriver crashing in Jenkins

### ISSUE THREAD

**Issue by gheeno** [serenity-bdd/serenity-core#1756]
Hi there can anyone point me to the right direction please?

OS : cat /etc/redhat-release
Red Hat Enterprise Linux Server release 7.6 (Maipo)

Chrome installed : 75.0.3770.100
ChromeDriver : 75.0.3770.140

Stacktrace : 

```
10:06:34 Aug 01, 2019 10:06:34 AM org.openqa.selenium.remote.DesiredCapabilities chrome
10:06:34 INFO: Using `new ChromeOptions()` is preferred to `DesiredCapabilities.chrome()`
10:06:34 Starting ChromeDriver 75.0.3770.140 (2d9f97485c7b07dc18a74666574f19176731995c-refs/branch-heads/3770@{#1155}) on port 16961
10:06:34 Only local connections are allowed.
10:06:34 Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
10:06:34 [1564675594.923][SEVERE]: bind() failed: Cannot assign requested address (99)
10:06:35 [2019-08-01 10:06:35,369] [WARN] Remote driver creation failed (unknown error: Chrome failed to start: exited abnormally), falling back on default driver creation
10:06:35 Starting ChromeDriver 75.0.3770.140 (2d9f97485c7b07dc18a74666574f19176731995c-refs/branch-heads/3770@{#1155}) on port 19986
10:06:35 Only local connections are allowed.
10:06:35 Please protect ports used by ChromeDriver and related test frameworks to prevent access by malicious code.
10:06:35 [1564675595.392][SEVERE]: bind() failed: Cannot assign requested address (99)
10:06:35 [2019-08-01 10:06:35,535] [ERROR] FAILED TO CREATE NEW WEBDRIVER_DRIVER INSTANCE class org.openqa.selenium.chrome.ChromeDriver: Could not instantiate new WebDriver instance of type class org.openqa.selenium.chrome.ChromeDriver (unknown error: Chrome failed to start: exited abnormally
10:06:35   (unknown error: DevToolsActivePort file doesn't exist)
10:06:35   (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
10:06:35 Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'
10:06:35 System info: host: 'fmdat1ap03.unix.ctcwest.ctc', ip: '192.168.241.234', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-957.21.3.el7.x86_64', java.version: '1.8.0_212'
10:06:35 Driver info: driver.version: ChromeDriver
10:06:35 remote stacktrace: #0 0x56308a7507a9 <unknown>
10:06:35 
10:06:35 net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate new WebDriver instance of type class org.openqa.selenium.chrome.ChromeDriver (unknown error: Chrome failed to start: exited abnormally
10:06:35   (unknown error: DevToolsActivePort file doesn't exist)
10:06:35   (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
10:06:35 Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'
10:06:35 System info: host: 'fmdat1ap03.unix.ctcwest.ctc', ip: '192.168.241.234', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-957.21.3.el7.x86_64', java.version: '1.8.0_212'
10:06:35 Driver info: driver.version: ChromeDriver
10:06:35 remote stacktrace: #0 0x56308a7507a9 <unknown>
10:06:35 
10:06:35 	at net.thucydides.core.webdriver.WebDriverFactory.newWebdriverInstance(WebDriverFactory.java:136)
10:06:35 	at net.thucydides.core.webdriver.WebDriverFacade.newDriverInstance(WebDriverFacade.java:153)
10:06:35 	at net.thucydides.core.webdriver.WebDriverFacade.newProxyDriver(WebDriverFacade.java:144)
10:06:35 	at net.thucydides.core.webdriver.WebDriverFacade.getProxiedDriver(WebDriverFacade.java:114)
10:06:35 	at net.thucydides.core.webdriver.WebDriverFacade.manage(WebDriverFacade.java:417)
10:06:35 	at com.fglsports.util.common.WebDriverUtil.setDriverTimeouts(WebDriverUtil.java:122)
10:06:35 	at com.fglsports.core.testconditions.PosTestConditions.beforeScenario(PosTestConditions.java:57)
10:06:35 	at com.fglsports.marks.TestConditions.lambda$beforeScenario$0(TestConditions.java:30)
10:06:35 	at com.fglsports.marks.TestConditions.safeMethodRun(TestConditions.java:53)
10:06:35 	at com.fglsports.marks.TestConditions.beforeScenario(TestConditions.java:30)
10:06:35 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
10:06:35 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
10:06:35 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
10:06:35 	at java.lang.reflect.Method.invoke(Method.java:498)
10:06:35 	at org.jbehave.core.steps.StepCreator$MethodInvoker.invoke(StepCreator.java:958)
10:06:35 	at org.jbehave.core.steps.StepCreator$BeforeOrAfterStep.perform(StepCreator.java:624)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$FineSoFar.run(PerformableTree.java:346)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$PerformableSteps.perform(PerformableTree.java:1090)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$ExamplePerformableScenario.perform(PerformableTree.java:1019)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$PerformableScenario.perform(PerformableTree.java:899)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$PerformableStory.performScenarios(PerformableTree.java:825)
10:06:35 	at org.jbehave.core.embedder.PerformableTree$PerformableStory.perform(PerformableTree.java:798)
10:06:35 	at org.jbehave.core.embedder.PerformableTree.performCancellable(PerformableTree.java:422)
10:06:35 	at org.jbehave.core.embedder.PerformableTree.perform(PerformableTree.java:393)
10:06:35 	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:292)
10:06:35 	at org.jbehave.core.embedder.StoryManager$EnqueuedStory.call(StoryManager.java:266)
10:06:35 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10:06:35 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
10:06:35 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
10:06:35 	at java.lang.Thread.run(Thread.java:748)
10:06:35 Caused by: org.openqa.selenium.WebDriverException: unknown error: Chrome failed to start: exited abnormally
10:06:35   (unknown error: DevToolsActivePort file doesn't exist)
10:06:35   (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)
10:06:35 Build info: version: '3.8.1', revision: '6e95a6684b', time: '2017-12-01T18:33:54.468Z'
10:06:35 System info: host: 'fmdat1ap03.unix.ctcwest.ctc', ip: '192.168.241.234', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-957.21.3.el7.x86_64', java.version: '1.8.0_212'
10:06:35 Driver info: driver.version: ChromeDriver
10:06:35 remote stacktrace: #0 0x56308a7507a9 <unknown>
```

**Comment by wakaleo** [comment]
It's a chrome driver issue of some kind.

**Comment by gheeno** [comment]
> It's a chrome driver issue of some kind.

Yeah it's odd since it is working locally. 
Mac OSX Mojave 
Chrome : Version 75.0.3770.142 (Official Build) (64-bit)
Chromedriver : 75.0.3770.140

**Comment by gheeno** [comment]
```10:28:57 [ERROR] Tests run: 18, Failures: 0, Errors: 1, Skipped: 10, Time elapsed: 16.316 s <<< FAILURE! - in com.fglsports.marks.AcceptanceTestSuite
10:28:57 [ERROR] When User opens backoffice v2 login page, user logs in(com.fglsports.marks.definitions.pos.backofficev2.LoginDefinitions)  Time elapsed: 0.665 s  <<< ERROR!
10:28:57 net.thucydides.core.webdriver.UnsupportedDriverException: Could not instantiate class org.openqa.selenium.chrome.ChromeDriver
10:28:57 	at com.loginTest(LoginDefinitions.java:17)
10:28:57 Caused by: net.thucydides.core.webdriver.UnsupportedDriverException: ```

**Comment by wakaleo** [comment]
Sometimes running on CI servers needs some extra chrome capabilities, e.g.

```
chrome.switches="""--start-maximized;--test-type;--no-sandbox;--ignore-certificate-errors;
                   --disable-popup-blocking;--disable-default-apps;--disable-extensions-file-access-check;
                   --incognito;--disable-infobars,--disable-gpu"""
```

**Comment by gheeno** [comment]
> Sometimes running on CI servers needs some extra chrome capabilities, e.g.
> 
> ```
> chrome.switches="""--start-maximized;--test-type;--no-sandbox;--ignore-certificate-errors;
>                    --disable-popup-blocking;--disable-default-apps;--disable-extensions-file-access-check;
>                    --incognito;--disable-infobars,--disable-gpu"""
> ```

I'll try this out - is it a comma separated?

**Comment by wakaleo** [comment]
That would go in the serenity.conf file - they are usually delimitated by semi-colons (I see there is a typo in the last line).

**Comment by gheeno** [comment]
> That would go in the serenity.conf file - they are usually delimitated by semi-colons (I see there is a typo in the last line).

Sorry it's me again; adding the `--headless` param works. Really odd that the non headless would make chrome crash. 

Side question : would the ` screenshots on failure ` annotation still work despite the tests run in headless?

**Comment by wakaleo** [comment]
Screenshots will still work in headless mode.

**Comment by gheeno** [comment]
Thank you so much - I think we can close this thread @wakaleo . I appreciate the help.

TLDR - use `--headless` in your `chrome.switch` settings.

**Comment by pankajrhel** [comment]
Hi,
Is there anyway we can update in ubuntu server side, so go Chrome don't try to open in GUI mode instead it will run in background.
My case once trying to open google-chrome-stable in cli its trying to open in graphic mode. as its CLI its getting crashed. That causing my jenkins jobs.

below is error message.

ERROR] Scenario: Verify New and Edited Location Note Owner, Date and Time.Scenario: Verify New and Edited Location Note Owner, Date and Time
12:35:21 [ERROR]   Run 1: launchApp() Chrome Browser Failed: unknown error: Chrome failed to start: crashed
12:35:21   (unknown error: DevToolsActivePort file doesn't exist)
12:35:21   (The process started from chrome location /usr/bin/google-chrome is no longer running, so ChromeDriver is assuming that Chrome has crashed.)

Can you please suggest to fix this. really struggling from past few days.

**Comment by globalworming** [comment]
as mentiones above, if you dont have a gui, use chrome options/parameters to start the browser in headless mode

**Comment by RahulBluehole** [comment]
hii in my case  i'm getting this error
net.thucydides.core.webdriver.DriverConfigurationError: Could not instantiate class org.openqa.selenium.chrome.ChromeDriver
Caused by: net.thucydides.core.webdriver.DriverConfigurationError: 
Could not instantiate new WebDriver instance of type class org.openqa.selenium.chrome.ChromeDriver (Could not start a new session. Response code 500. Message: unknown error: DevToolsActivePort file doesn't exist
Build info: version: '4.1.2', revision: '9a5a329c5a'
System info: host: 'bluehole-Lenovo-V15-G2-ITL-Ua', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '5.15.0-46-generic', java.version: '9.0.4'
Driver info: org.openqa.selenium.chrome.ChromeDriver
Command: [null, newSession {capabilities=[Capabilities {acceptInsecureCerts: false, browserName: chrome, goog:chromeOptions: {args: [--start-maximized, --test-type, --no-sandbox, --ignore-certificate-errors, \n                         ..., --disable-default-apps, --disable-extensions-file-a..., \n                         ..., --disable-infobars, --disable-gpu, --headless], extensions: []}}], desiredCapabilities=Capabilities {acceptInsecureCerts: false, browserName: chrome, driver: chrome, goog:chromeOptions: {args: [--start-maximized, --test-type, --no-sandbox, --ignore-certificate-errors, \n                         ..., --disable-default-apps, --disable-extensions-file-a..., \n                         ..., --disable-infobars, --disable-gpu, --headless], extensions: []}}}]). See below for more details.
Caused by: org.openqa.selenium.SessionNotCreatedException: 
Could not start a new session. Response code 500. Message: unknown error: DevToolsActivePort file doesn't exist

**Comment by RahulBluehole** [comment]
Im using ubantu system with cucumber serenity.conf file
drivers {
  windows {
    webdriver.chrome.driver = src/test/resources/webdriver/chromedriver.exe
  }
  linux {
    webdriver.chrome.driver = src/test/resources/web_driver/linux/chromedriver
  }
  timeouts {
      implicitlywait = 10000
      fluentwait = 30000
    }
}
environments {
   prod {
        webdriver.base.url = "https://www.card.ai/"
    }
    euro {
        webdriver.base.url = "https://qa.eu.com/"
    }
    web {
        webdriver.base.url = "https://web.com/"
    }
    webkorps {
        webdriver.base.url = "https://qa .com/"
    }
}
webdriver {
 driver = chrome
}
  headless.mode = true

serenity {
    serenity.restart.browser.for.each = never
    take.screenshots = FOR_FAILURES
}
#
# Chrome options can be defined using the chrome.switches property
#
chrome.switches = """--start-maximized;--test-type;--no-sandbox;--ignore-certificate-errors;
                                     --disable-popup-blocking;--disable-default-apps;--disable-extensions-file-access-check;
                                     --incognito;--disable-infobars;--disable-gpu"""
